lib/gnat.ex

```diff
@@ -273,6 +353,15 @@ defmodule Gnat do
     %{state | receivers: receivers}
   end
 
+  defp process_message({:msg, topic, @request_sid, reply_to, body}, state) do
+    if Map.has_key?(state.request_receivers, topic) do
+      send state.request_receivers[topic], {:msg, %{topic: topic, body: body, reply_to: reply_to, gnat: self()}}
+      state
+    else
+      Logger.error "#{__MODULE__} got a response for a request, but that is no longer registered"
```

Comment:
What do you think about the case where someone publishes a response to a request after we've timed out? Normally we wouldn't receive the message because we've unsub'd, but now that we're using a shared wildcard subscription for all requests, this code path isn't really an error anymore.

Looking at the go-nats client, they drop any message that doesn't match:
1. https://github.com/nats-io/go-nats/blob/d4841f2d6a3ab7745d940768bfbc2fe9d60c65ed/nats.go#L2140-L2164
2. Example showing similar code ignores unknown response messages without raising an error: https://play.golang.org/p/3pVFp3b8O3G

Reply:
:+1: that sounds like a good behavior to me. No reason to log an additional error if we do eventually receive a response, we already would have returned an error indication to the user when the timeout occured